src/library-aux/scala/Any.scala

```diff
@@ -138,13 +138,87 @@ abstract class Any {
    */
   final def isInstanceOf[T0]: Boolean = sys.error("isInstanceOf")
 
-  /** Cast the receiver object to be of type `T0`.
-   *
-   *  Note that the success of a cast at runtime is modulo Scala's erasure semantics.
-   *  Therefore the expression `1.asInstanceOf[String]` will throw a `ClassCastException` at
-   *  runtime, while the expression `List(1).asInstanceOf[List[String]]` will not.
-   *  In the latter example, because the type argument is erased as part of compilation it is
-   *  not possible to check whether the contents of the list are of the requested type.
+  /** Tell the compiler to skip type safety and treat the receiver object to be of type `T0`.
```

Comment:
We should absolutely preserve the word "cast" in the title of this Scaladoc. "casting" is a technical word with a specific meaning, which we use not only in Scala but in other languages. Hiding it is doing a disservice to those who already know the word:
```suggestion
  /** Cast the receiver object to be of type `T0`.
   *
   *  Casting drops type safety and treats the receiver object to be of type `T0`,
   *  without any checking at compile-time nor (comprehensive) checking at run-time.
```

Reply:
As we have both agreed on, converting between primitive types is not a cast. Already for this reason we should not mention cast or at least not alone and include conversion in the title as well. Also, as far as I can tell people coming from Java tend to believe that casting to a non-primitive type always involves type checking (see my answer to your distinction between type testing/checking). IMO it doesn't really matter that this is a misconception of casting, I see that it is a pitfall and with this PR I try to improve in this regard. I still mention cast and conversion in the description afterwards, IMO good enough

Reply:
I don't have a strong opinion either way here, but I find Rob's argument persuasive enough.